pkg/observ/observ.go

```diff
@@ -17,16 +18,62 @@ type observabilityContext struct {
 	spanCtx context.Context
 }
 
-// RegisterTraceExporter returns a jaeger exporter for exporting traces to opencensus.
+// jaegerTraceExporter is used to hold the trace exporter for Jaeger
+type jaegerTraceExporter struct {
+	exporter (*(jaeger.Exporter))
+}
+
+// datadogTraceExporter is used to hold the trace exporter for Datadog
+type datadogTraceExporter struct {
+	exporter (*(datadog.Exporter))
+}
+
+// Exporter might be a JaejerExporter, DataDogExporter, etc
+// All Exporters *must* implement these two functions.
+// Flush method stops the trace exporter, this method needs to be
+// called for most of the TraceExporters using opencensus
+type Exporter interface {
+	RegisterTraceExporter(URL, service, ENV string) error
+	Flush()
+}
+
+// RetrieveTracer determines the type of TraceExporter service for exporting traces from opencensus
+// User can choose from multiple tracing services (datadog, jaegar)
+// TODO: Extend beyond jaeger and datadog
+func RetrieveTracer(TraceExporter string) (Exporter, error) {
+
+	const op errors.Op = "RegisterTracer"
+
+	switch TraceExporter {
+	case "jaeger":
+		return jaegerTraceExporter{}, nil
+	case "datadog":
+		return datadogTraceExporter{}, nil
+	default:
+		return nil, errors.E(op, "Exporter not specified. Traces won't be exported")
```

Comment:
let's differentiate between an empty string, and an unsupported exporter. This way users know that if they put something like "stackdriver", that it's still not supported.